docker/cluster-test/cluster-test.Dockerfile

```diff
@@ -18,7 +18,9 @@ RUN rustup install $(cat rust-toolchain)
 COPY . /libra
 RUN docker/cluster-test/compile.sh
 FROM debian:buster
-RUN apt-get update && apt-get install -y openssh-client curl && curl -L "https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl" -o /usr/local/bin/kubectl && chmod +x /usr/local/bin/kubectl
+RUN apt-get update && apt-get install -y openssh-client curl
+# Use retries to download kubectl
+RUN ["bash", "-c", "download_kubectl() {   local location=/usr/local/bin/kubectl;   for ((i = 0; i < 30; i++)); do     curl -L \"https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl\" -o ${location};     if [ $? -gt 0 ]; then       echo \"curl failed. retrying in 10 secs.\";       sleep 10;     else       chmod +x ${location};       echo \"successfully installed kubectl\";       return 0;     fi;   done;   return 1; }; download_kubectl"]
```

Comment:
I feel like it would be cleaner to do the retry logic in the build script instead of inside the Dockerfile.  We can keep Dockerfile as simple as possible.

Reply:
Another thought, can we leverage DLC to avoid re-downloading kubectl altogether if there is no version change? For instance, this block can be moved to before line 15.

Reply:
Makes sense. Will do.

Reply:
> Another thought, can we leverage DLC to avoid re-downloading kubectl altogether

We are already leveraging DLC. This is using the builder pattern by building libra in a separate container. Its confusing because there is no newline b/w the two builds. I have added one.